meterfs/meterfs.c

```diff
@@ -667,6 +667,9 @@ int meterfs_devctl(meterfs_i_devctl_t *i, meterfs_o_devctl_t *o, meterfs_ctx_t *
 			if (SECTORS(&h, ctx->sectorsz) > i->allocate.sectors)
 				return -EINVAL;
 
+			if (i->allocate.sectors * ctx->sectorsz >= ctx->sz)
+				return -ENOMEM;
+
```

Comment:
We should check how many sectors are free, not how many sectors are in total.

Reply:
done